compiler/qsc_formatter/src/formatter.rs

```diff
@@ -121,36 +133,124 @@ fn apply_rules(
     left: &ConcreteToken,
     whitespace: &str,
     right: &ConcreteToken,
-    code: &str,
-    indent_level: usize,
+    state: &mut FormatterState,
```

Comment:
Minor style nit: I think once you have a parameter named `state`, you're basically writing the `impl` for that struct. It would seem more intuitive to me, if instead of being a free function, `apply_rules` was a method of `FormatterState` (or just `Formatter`) and took `&mut self` instead of `state` as a named parameter.

Reply:
`apply_rules` as well as the new state updater functions are put into the new `impl` for the renamed `Formatter` struct. Thanks for the suggestion!